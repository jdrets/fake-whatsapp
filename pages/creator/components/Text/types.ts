export interface TextTypes {
  onDelete: () => void
  text: string
  owner: string
  isFirstMessage: boolean
}

export interface WrapperTypes {
  owner: string
  isFirstMessage?: boolean
}
