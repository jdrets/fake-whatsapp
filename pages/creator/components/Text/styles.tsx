import styled from 'styled-components'

import { WrapperTypes } from './types'

export const Wrapper = styled.div<WrapperTypes>`
  position: relative;
  display: flex;
  justify-content: ${({ owner }) => owner === 'SENDER' ? 'flex-end' : 'flex-start'};
`

const getTriangle = (owner: string, isFirstMessage: boolean) => {
  if (owner === 'SENDER' && isFirstMessage) {
    return '12px 0 12px 12px'
  }
  if (owner === 'RECEIVER' && isFirstMessage) {
    return '0 12px 12px 12px'
  }
  return '12px'
}

export const TextWrapper = styled.div<WrapperTypes>`
  position: relative;
  border-radius: ${({ owner, isFirstMessage }) => getTriangle(owner, isFirstMessage)};
  padding: 8px 12px;
  text-align: left;
  font-size: 16px;
  background: ${({ owner }) => owner === "SENDER" ? '#dafebe' : 'white'};
  margin-bottom: 2px;
  margin-top: ${({ isFirstMessage }) => isFirstMessage && '6px'};
  min-width: 120px;
  max-width: 90%;
  border-bottom: 1px solid #dbdbdb;
`

export const TextLabel = styled.div`
  padding-right: 68px;
  line-height: 18px;
`

export const HourWrapper = styled.div`
  position: absolute;
  right: 8px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  opacity: 0.4;
  margin-top: -14px;
`

export const Hour = styled.span`
  font-size: 12px;
`

export const SenderTriangle = styled.div`
  clip-path: polygon(100% 0, 0% 100%, 0 0);
  background: #dafebe;
  height: 10px;
  width: 8px;
  position: absolute;
  right: -8px;
  top: 6px
`

export const ReceiverTriangle = styled.div`
  clip-path: polygon(100% 1%, 0 0, 100% 100%);
  background: white;
  height: 10px;
  width: 8px;
  position: absolute;
  left: -8px;
  top: 6px
`
