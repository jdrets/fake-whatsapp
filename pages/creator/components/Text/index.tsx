import React, {FunctionComponent} from 'react'

import {Wrapper, TextWrapper, HourWrapper, Hour, TextLabel, SenderTriangle, ReceiverTriangle} from './styles'
import {TextTypes} from './types'
import Tick from './components/Tick'

const Text: FunctionComponent<TextTypes> = ({onDelete, owner, text, isFirstMessage}) => {
  const isReceiver = owner === 'RECEIVER'
  const isSender = owner === 'SENDER'

  return (
    <Wrapper onClick={onDelete} owner={owner}>
      {(isReceiver && isFirstMessage) && <ReceiverTriangle/>}
      <TextWrapper owner={owner} isFirstMessage={isFirstMessage}>
        <TextLabel>{text}</TextLabel>
        <HourWrapper>
          <Hour>13:30</Hour>
          {isSender && <Tick/>}
        </HourWrapper>
      </TextWrapper>
      {(isSender && isFirstMessage) && <SenderTriangle/>}
    </Wrapper>
  )
}

export default Text
