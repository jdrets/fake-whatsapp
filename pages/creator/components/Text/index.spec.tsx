import React from 'react'
import { render } from '@testing-library/react'

import Text from '.'

const setup = (mock = {}) => (
  render(
    <Text
      onDelete={jest.fn()}
      owner="RECEIVER"
      text="dummy text"
      isFirstMessage
      hour="13:00"
      {...mock}
    />
  )
)

describe('<Text />', () => {
  test('Should match with a snapshot when owner is RECEIVER', () => {
    const { container } = setup({
      owner: 'RECEIVER'
    })

    expect(container).toMatchSnapshot()
  })

  test('Should match with a snapshot when owner is SENDER', () => {
    const { container } = setup({
      owner: 'SENDER'
    })

    expect(container).toMatchSnapshot()
  })

  test('Should match with a snapshot when isFirstMessage is false', () => {
    const { container } = setup({
      owner: 'SENDER',
      isFirstMessage: false
    })

    expect(container).toMatchSnapshot()
  })
})
