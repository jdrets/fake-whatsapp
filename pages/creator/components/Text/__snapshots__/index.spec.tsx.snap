// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Should match with a snapshot when isFirstMessage is false 1`] = `
<div>
  <div
    class="styles__Wrapper-sc-1any8y4-0 kJrgLD"
  >
    <div
      class="styles__TextWrapper-sc-1any8y4-1 jXxrWW"
    >
      <div
        class="styles__TextLabel-sc-1any8y4-2 kjuqUo"
      >
        dummy text
      </div>
      <div
        class="styles__HourWrapper-sc-1any8y4-3 cfNxXb"
      >
        <span
          class="styles__Hour-sc-1any8y4-4 eSNUDC"
        >
          13:00
        </span>
        <svg
          height="14px"
          style="margin-left: 0px;"
          version="1.1"
          viewBox="0 0 594.149 594.149"
          width="14px"
        >
          <g>
            <g
              id="done-all"
            >
              <path
                d="M448.8,161.925l-35.7-35.7l-160.65,160.65l35.7,35.7L448.8,161.925z M555.899,126.225l-267.75,270.3l-107.1-107.1 l-35.7,35.7l142.8,142.8l306-306L555.899,126.225z M0,325.125l142.8,142.8l35.7-35.7l-142.8-142.8L0,325.125z"
              />
            </g>
          </g>
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Text /> Should match with a snapshot when owner is RECEIVER 1`] = `
<div>
  <div
    class="styles__Wrapper-sc-1any8y4-0 cDpuqU"
  >
    <div
      class="styles__ReceiverTriangle-sc-1any8y4-6 guGaBl"
    />
    <div
      class="styles__TextWrapper-sc-1any8y4-1 bOBFgJ"
    >
      <div
        class="styles__TextLabel-sc-1any8y4-2 kjuqUo"
      >
        dummy text
      </div>
      <div
        class="styles__HourWrapper-sc-1any8y4-3 cfNxXb"
      >
        <span
          class="styles__Hour-sc-1any8y4-4 eSNUDC"
        >
          13:00
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Text /> Should match with a snapshot when owner is SENDER 1`] = `
<div>
  <div
    class="styles__Wrapper-sc-1any8y4-0 kJrgLD"
  >
    <div
      class="styles__TextWrapper-sc-1any8y4-1 lfLzFY"
    >
      <div
        class="styles__TextLabel-sc-1any8y4-2 kjuqUo"
      >
        dummy text
      </div>
      <div
        class="styles__HourWrapper-sc-1any8y4-3 cfNxXb"
      >
        <span
          class="styles__Hour-sc-1any8y4-4 eSNUDC"
        >
          13:00
        </span>
        <svg
          height="14px"
          style="margin-left: 0px;"
          version="1.1"
          viewBox="0 0 594.149 594.149"
          width="14px"
        >
          <g>
            <g
              id="done-all"
            >
              <path
                d="M448.8,161.925l-35.7-35.7l-160.65,160.65l35.7,35.7L448.8,161.925z M555.899,126.225l-267.75,270.3l-107.1-107.1 l-35.7,35.7l142.8,142.8l306-306L555.899,126.225z M0,325.125l142.8,142.8l35.7-35.7l-142.8-142.8L0,325.125z"
              />
            </g>
          </g>
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
        </svg>
      </div>
    </div>
    <div
      class="styles__SenderTriangle-sc-1any8y4-5 kOEvHj"
    />
  </div>
</div>
`;
